handler/oauth2/flow_revoke.go

```diff
@@ -0,0 +1,26 @@
+package oauth2
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/ory-am/fosite"
+)
+
+type TokenRevocationHandler struct {
+	TokenRevocationStorage TokenRevocationStorage
+}
+
+// RevokeToken implements https://tools.ietf.org/html/rfc7009#section-2.1
+func (r *TokenRevocationHandler) RevokeToken(ctx context.Context, token string, tokenType fosite.TokenType) error {
+	if tokenType == fosite.RefreshToken {
```

Comment:
It looks like this is clashing with the definition:

>   token_type_hint  OPTIONAL.  A hint about the type of the token
           submitted for revocation.  Clients MAY pass this parameter in
           order to help the authorization server to optimize the token
           lookup.  If the server is unable to locate the token using
           the given hint, it MUST extend its search across all of its
           supported token types.  An authorization server MAY ignore
           this parameter, particularly if it is able to detect the
           token type automatically.  This specification defines two
           such values: